BadParcelableException when using Proguard

This is easily avoided with a Proguard rule, so adding it to the documentation may be enough.

The issue I had was getting a BadParcelableException when using Proguard, this is the stack trace:

> FATAL EXCEPTION: main
>  Process: com.example.app, PID: 15577
>  java.lang.RuntimeException: Unable to start service com.firebase.jobdispatcher.GooglePlayReceiver@c883f90 with Intent { act=com.google.android.gms.gcm.ACTION_TASK_READY cmp=com.example.app/com.firebase.jobdispatcher.GooglePlayReceiver (has extras) }: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: com.google.android.gms.gcm.PendingCallback
>      at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3027)
>      at android.app.ActivityThread.-wrap17(ActivityThread.java)
>      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1442)
>      at android.os.Handler.dispatchMessage(Handler.java:102)
>      at android.os.Looper.loop(Looper.java:148)
>      at android.app.ActivityThread.main(ActivityThread.java:5417)
>      at java.lang.reflect.Method.invoke(Native Method)
>      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
>      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
>   Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: com.google.android.gms.gcm.PendingCallback
>      at android.os.Parcel.readParcelableCreator(Parcel.java:2411)
>      at android.os.Parcel.readParcelable(Parcel.java:2337)
>      at android.os.Parcel.readValue(Parcel.java:2243)
>      at android.os.Parcel.readArrayMapInternal(Parcel.java:2592)
>      at android.os.BaseBundle.unparcel(BaseBundle.java:221)
>      at android.os.Bundle.getParcelable(Bundle.java:786)
>      at com.firebase.jobdispatcher.f.a(SourceFile:39)
>      at com.firebase.jobdispatcher.GooglePlayReceiver.a(SourceFile:114)
>      at com.firebase.jobdispatcher.GooglePlayReceiver.onStartCommand(SourceFile:82)
>      at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3010)
>      at android.app.ActivityThread.-wrap17(ActivityThread.java) 
>      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1442) 
>      at android.os.Handler.dispatchMessage(Handler.java:102) 
>      at android.os.Looper.loop(Looper.java:148) 
>      at android.app.ActivityThread.main(ActivityThread.java:5417) 
>      at java.lang.reflect.Method.invoke(Native Method) 
>      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
>      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 

Could be related to #18 

This is fixed by adding the following Proguard rule:

`-keep class com.google.android.gms.** { *; }`

--- a/jobdispatcher/build.gradle
+++ b/jobdispatcher/build.gradle
@@ -24,6 +24,7 @@ android {
         }
         release {
             minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt')
         }
     }
 

